examples/stm32/nucleo-h743zi-baremetal/main.c

```diff
@@ -1,67 +1,74 @@
 // Copyright (c) 2023 Cesanta Software Limited
 // All rights reserved
 
-#include "mcu.h"
+#include "hal.h"
 #include "mongoose.h"
 
-#define LED1 PIN('B', 0)      // On-board LED pin (green)
-#define LED2 PIN('E', 1)      // On-board LED pin (yellow)
-#define LED3 PIN('B', 14)     // On-board LED pin (red)
-#define BLINK_PERIOD_MS 1000  // LED blinking period in millis
+#define LED1 PIN('B', 0)   // On-board LED pin (green)
+#define LED2 PIN('E', 1)   // On-board LED pin (yellow)
+#define LED3 PIN('B', 14)  // On-board LED pin (red)
 
-static uint64_t s_ticks;
-void SysTick_Handler(void) { s_ticks++; }
-uint64_t mg_millis(void) { return s_ticks; }
+#define LED LED2              // Use blue LED for blinking
+#define BLINK_PERIOD_MS 1000  // LED blinking period in millis
 
-static void timer_cb(void *arg) {
-  gpio_toggle(LED2);                      // Blink LED
-  MG_INFO(("ticks: %lld", mg_millis()));  // Log something
-  (void) arg;
+static uint64_t s_ticks;      // Milliseconds since boot
+void SysTick_Handler(void) {  // SyStick IRQ handler, triggered every 1ms
+  s_ticks++;
 }
 
-int main(void) {
-  clock_init();
-  systick_init(CPU_FREQUENCY / 1000);  // Increment s_ticks every ms
-  gpio_output(LED2);                   // Setup LED
-  uart_init(UART_DEBUG, 115200);       // Initialise debug printf
+uint64_t mg_millis(void) {  // Let Mongoose use our uptime function
+  return s_ticks;           // Return number of milliseconds since boot
+}
 
-  char rev = chiprev();
-  MG_INFO(("Chip revision: %c, max cpu clock: %u MHz", rev, (rev == 'V') ? 480 : 400));
```

Comment:
You need to know the chip revision, revY does not run at 480MHz but at 400MHz, that is why I put that there

Reply:
Don't we show the CPU clock here?

```c
  MG_INFO(("Starting, CPU freq %g MHz", (double) SystemCoreClock / 1000000));
```

If the SystemCoreClock holds a wrong number, then it is SystemInit() that should be fixed insted.

Reply:
I mean, if the chip does some weirdo crashes, the user will see a nice message before that... of if it does not boot, a clue to search why.

The function does not change freq according to chip revno, as the PLL numbers are fixed enums decided at compile time.

Reply:
Restored!